### PR TITLE
Updates CMakeLists.txt file to compile on Setonix.

### PR DESCRIPTION
**Pull request by dipietrantonio**

### CODE REVIEW

**Review by bwmeyers — COMMENTED** [review]
Couple of comments. But otherwise, looking okay for draft stages.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,38 @@
...
 #cmake_policy(SET CMP0012 NEW)
 #cmake_policy(SET CMP0048 NEW)
 
+# Define CMake options to choose the GPU flavour
+option(USE_CUDA "Compile the code with NVIDIA GPU support." OFF)
+option(USE_HIP "Compile the code with AMD GPU support." OFF)
+
+# Find packages needed
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
+#find_package(CUDA REQUIRED)
```
[the remark this answers is not in the document]
Comment (bwmeyers) [reply CMakeLists.txt:12]: Not needed (since Cmake supports CUDA as a language). But we may still test it on other envs.

**Review by bwmeyers — APPROVED** [review]
I think this looks good. I added some extra comments and tweaked some of the messaging/output. Feel free to have a look.

**Comment by dipietrantonio** [comment]
Looks good to me. Merge it when you want.

**Pull request merged**